newIDE/app/src/GameDashboard/GameAnalyticsPanel.js

```diff
@@ -2,53 +2,87 @@
+import { formatISO, subDays } from 'date-fns';
+import { Column, Line } from '../UI/Grid';
 import { type Game } from '../Utils/GDevelopServices/Game';
 import { ColumnStackLayout } from '../UI/Layout';
 import Text from '../UI/Text';
 import {
   type GameMetrics,
-  getGameMetrics,
+  getGameMetricsFrom,
 } from '../Utils/GDevelopServices/Analytics';
 import AuthenticatedUserContext from '../Profile/AuthenticatedUserContext';
 import PlaceholderError from '../UI/PlaceholderError';
 import SelectField from '../UI/SelectField';
 import SelectOption from '../UI/SelectOption';
-import { CircularProgress } from '@material-ui/core';
-import { Table, TableBody, TableRow, TableRowColumn } from '../UI/Table';
 import AlertMessage from '../UI/AlertMessage';
-import subDays from 'date-fns/subDays';
-import { type PublicGame } from '../Utils/GDevelopServices/Game';
+import { ResponsiveLineStackLayout } from '../UI/Layout';
+import {
+  ResponsiveContainer,
+  LineChart,
+  Line as RechartsLine,
+  XAxis,
+  YAxis,
+  Tooltip,
+  CartesianGrid,
+  AreaChart,
+  Area,
+} from 'recharts';
+import GDevelopThemeContext from '../UI/Theme/ThemeContext';
+import PlaceholderLoader from '../UI/PlaceholderLoader';
+import {
+  buildChartData,
+  daysShownForYear,
+  durationValues,
+} from './GameAnalyticsEvaluator';
+
+const chartMargins = {
+  top: 5,
+  bottom: 5,
+  right: 25,
+  left: 0,
+};
+// There is a known bug with recharts that causes the chart to not render if the width is 100%
+// in a flexbox component. check https://github.com/recharts/recharts/issues/172
+const chartWidth = '99%';
+const chartHeight = 300;
+
+const minutesFormatter = value => {
+  return value.toFixed(2);
+};
 
-const styles = {
-  tableRowStatColumn: {
-    width: 100,
-  },
+const percentFormatter = value => {
+  return value.toFixed(2);
 };
 
 type Props = {|
   game: Game,
-  publicGame: ?PublicGame,
 |};
 
-export const GameAnalyticsPanel = ({ game, publicGame }: Props) => {
+export const GameAnalyticsPanel = ({ game }: Props) => {
   const { getAuthorizationHeader, profile } = React.useContext(
     AuthenticatedUserContext
   );
-  const [gameRollingMetrics, setGameMetrics] = React.useState<?GameMetrics>(
+
+  const [gameRollingMetrics, setGameMetrics] = React.useState<?(GameMetrics[])>(
     null
   );
+  const { yearChartData, monthChartData } = React.useMemo(
+    () => buildChartData(gameRollingMetrics),
+    [gameRollingMetrics]
+  );
+  const [dataPeriod, setDataPeriod] = React.useState('month');
+  const chartData = dataPeriod === 'year' ? yearChartData : monthChartData;
+
   const [gameRollingMetricsError, setGameMetricsError] = React.useState<?Error>(
     null
   );
   const [isGameMetricsLoading, setIsGameMetricsLoading] = React.useState(false);
 
-  const yesterdayIsoDate = formatISO(subDays(new Date(), 1), {
+  // TODO In some timezones, it might ask one less or extra day.
+  const lastYearIsoDate = formatISO(subDays(new Date(), daysShownForYear), {
```

Comment:
I think it could be ok as it will only do some imprecisions at the end of the year.

Reply:
Sounds ok, could we still force a timezone?